File: 240322/가장 많이 겹치는 구간/section-with-maximum-overlap.java
import java.util.Scanner;

public class Main {
    private static final Scanner sc = new Scanner(System.in);
    private static final int MAX_NUMBER = 200000;

    public static void main(String[] args) {
        int n = sc.nextInt();
        int[] checked = new int[MAX_NUMBER + 1];

        for (int i = 0; i < n; i++) {
            int s = sc.nextInt();
            int e = sc.nextInt();

            checked[s]++;
            checked[e]--;
        }

        int[] sum = new int[MAX_NUMBER + 1];
        for (int i = 1; i <= MAX_NUMBER; i++) {
            sum[i] = sum[i - 1] + checked[i];
        }

        int answer = -1;
        for (int i = 1; i <= MAX_NUMBER; i++) {
            answer = Math.max(answer, sum[i]);
        }
        System.out.println(answer);
    }
}

File: 240426/가장 많이 겹치는 구간/section-with-maximum-overlap.kt
import java.util.*
import kotlin.math.*

val sc = Scanner(System.`in`)

val n = sc.nextInt()
val weights = IntArray(200001)

fun main() {
    for (i in 1..n) {
        val x1 = sc.nextInt()
        val x2 = sc.nextInt()

        weights[x1]++
        weights[x2]--
    }

    var answer = 0
    var sum = 0
    for (i in 1..weights.size-1) {
        sum += weights[i]
        answer = max(answer, sum)
    }
    
    print(answer)
}